bin/rose-make-docs

```diff
@@ -0,0 +1,176 @@
+#
+# OPTIONS
+#     --venv
+#         Build virtualenv for temporarilly installing python dependencies
+#         if necessary.
+#     --dev
+#         Development mode, don't remove virtualenv after build.
+#
+# ENVIRONMENT VARIABLES
+#     SPHINX_DEV_MODE
+#         Set to `true` in order to make the `rose test-battery` command use
+#         development mode.
+#
+# BUILD
+#     The format(s) to build the documentation in - default html.
+#     Avaliable formats are listed in the sphinx documentation
+#     (http://www.sphinx-doc.org/en/stable/builders.html).
+#     The most commonly used formats are:
+#
+#     `html`
+#         For building standalone HTML files.
+#     `singlehtml`
+#         For building a single page HTML document.
+#     `latexpdf`
+#         For building PDF documentation.
+#
+# DEVELOPMENT BUILDS
+#     For development purposes use the following BUILDs:
+#
+#     `doctest`
+#         Runs any doctest examples present in documented python modules.
+#     `strict`
+#         Runs sphinx-build in a "strict" mode where cashing is turned off and
+#         warnings are turned into errors.
+#     `linkcheck`
+#         Checks external links.
+#
+#-----------------------------------------------------------------------------
+set -e
+set -o pipefail
+
+# Move into rose directory
+cd "$(dirname $0)/../"
+
+VENV_PATH='venv'  # Path for virtualenv.
+USING_VENV=false  # Set to `true` when the virtualenv is being used.
+SPHINX_PATH=sphinx  # Path to the sphinx directory.
+
+venv-activate () {
+    USING_VENV=true
+    . "${VENV_PATH}/bin/activate"
+}
+
+venv-install () {
+    venv-destroy
+    virtualenv --python=python2.7 "${VENV_PATH}"
+    venv-activate
+    pip install sphinx
+    pip install sphinx_rtd_theme
+}
+
+venv-deactivate () {
+    deactivate >/dev/null 2>&1 || true
+}
+
+venv-destroy () {
+    venv-deactivate
+    rm -rf "${VENV_PATH}"
+}
+
+# Parse command line args.
+VENV_MODE=false
+DEV_MODE=${SPHINX_DEV_MODE:=false}
+BUILDS=''
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        --venv)
+            VENV_MODE=true
+            shift
+            ;;
+        --dev)
+            DEV_MODE=true
+            shift
+            ;;
+        *)
+            BUILDS="$BUILDS $1"
+            shift
+            ;;
+    esac
+done
+if [[ -z "${BUILDS}" ]]; then
+    BUILDS='html'
+fi
+
+# Use virtualenv if present and requested or if we are in development mode.
+if [[ "${DEV_MODE:-false}" || "${VENV_MODE:-false}" ]]; then
+    if [ -d "${VENV_PATH}" ]; then
+        venv-activate
+    fi
+fi
+
+# Check optional builders.
+for build in ${BUILDS}; do
+    if ! rose-check-software-docs ${build} >/dev/null; then
+        echo "Software dependencies for a ${build} build are missing."
+        echo "run `rose software-check-docs ${build}` for details."
+        exit 1
+    fi
+done
+
+# Check core (sphinx) builder.
+if ! rose-check-software-docs core >/dev/null; then
+    if ${VENV_MODE}; then
```

Comment:
I've not used quotes here as `VENV_MODE` is a constant which is set to either `true` or `false`, quotes defend against spaces which be erroneous if present.